internal/test_cases/file_search_test_case.go

```diff
@@ -0,0 +1,81 @@
+package test_cases
+
+import (
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+type FileSearchTestCase struct {
+	Pattern          string
+	FilePaths        []string
+	ExpectedExitCode int
+	ExpectedOutput   []string
+	Recursive        bool
+}
+
+func (testCase FileSearchTestCase) Run(stageHarness *test_case_harness.TestCaseHarness) error {
+	return runFileSearchTestCases([]FileSearchTestCase{testCase}, stageHarness)
+}
+
+type FileSearchTestCases []FileSearchTestCase
```

Comment:
```suggestion
type FileSearchTestCaseCollection []FileSearchTestCase
```

Prefer to use nouns over plural form -- makes it clearer that you're dealing with a separate object/struct vs. just an array of test cases.